src/App.tsx

```diff
@@ -14,210 +34,76 @@ export const App: React.FC = () => {
     }
   }, []);
 
+  const loadTodos = useCallback(async () => {
+    if (!user) {
+      return;
+    }
+
+    try {
+      setErrorMessage('');
+      const loadedTodos = await getTodos(user.id);
+
+      setTodos(loadedTodos);
+    } catch (error) {
+      setErrorMessage('Something went wrong while loading todos');
+    }
+  }, []);
+
+  useEffect(() => {
+    loadTodos();
+  }, []);
+
+  const filteredTodos = useMemo(() => {
+    return todos.filter(todo => {
+      switch (filterType) {
+        case FilterType.All:
+          return todo;
+
+        case FilterType.Active:
+          return !todo.completed;
+
+        case FilterType.Completed:
+          return todo.completed;
+
+        default:
+          throw new Error('Invalid type');
+      }
+    });
```

Comment:
If you will do a switch before filtering, the filter type will be checked only once, if you make it inside the filter method, the filter type will be checked on every element during the iteration.
```suggestion
  const filteredTodos = useMemo(() => {
    switch (filterType) {
      case FilterType.Active:
        return todos.filter(todo => (
          !todo.completed
        )
      ............
    }
  });
```